### ISSUE TITLE
This extension may soon no longer be supported because it doesn't follow best practices for Chrome extensions.

### ISSUE THREAD

**Issue by michaelkrieger** [bogenpirat/remote-torrent-adder#353]
Chrome says "This extension may soon no longer be supported because it doesn't follow best practices for Chrome extensions. ([Learn more](https://support.google.com/chrome_webstore/answer/2664769))"

Needs an update soon, I would assume?

**Comment by bogenpirat** [comment]
i guess there's still the option to manually re-enable it then... but yeah, this has been coming for quite some time and i guess i should tend to migrating it to v3 at some point. last i checked, this is not a trivial task, so it'll take some resources. i'll update this issue once i'm done

**Comment by SiriosDev** [comment]
news?

**Comment by Elegant996** [comment]
You could also assist with migrating instead of asking for an update. There are tons of browsers (other than Chrome!) that support manifest v2. I tried to do the migration myself on a fork but it's much more daunting than you'd expect. They really do not want us loading multiple scripts simultaneously. This may mean the addon needs to become more modular.

**Comment by off3nc3** [comment]
> i guess there's still the option to manually re-enable it then... but yeah, this has been coming for quite some time and i guess i should tend to migrating it to v3 at some point. last i checked, this is not a trivial task, so it'll take some resources. i'll update this issue once i'm done

Please don't le this extension die , it was the best thing ever and especially for Chrome users ! I have found zero alternatives as of now without it.

**Comment by stein3x** [comment]
The extension no longer works , its not loading the menu for options or anything. Is there any chance of updating since i didn't find any alternative and its the only one that worked flawlessly

**Comment by bogenpirat** [comment]
at least as of chrome 131 i can still re-enable it via chrome://extensions/

**Comment by Feriman22** [comment]
The re-enable works with Chrome v133

**Comment by off3nc3** [comment]
> The re-enable works with Chrome v133

Problem is i can't because i no longer have it in my extension folder in appdata for some unknown reason because i never deleted it from the appdata folder but w/e, so that's a no go.

We really need @bogenpirat to migrate this thing to v3 , if ever.

**Comment by off3nc3** [comment]
> i mean, you don't (yet) _technically_ need me too, considering you can still just re-enable to extension, be it the chrome web store version or the unpacked version. in the wiki section of this repo, there're even instructions how to migrate your settings to a version that runs unzipped, in case you'd have too much to reconfigure.

Appreciate you taking the time to send me a reply , for some unknown reason it was missing from my folder  : %localappdata%\Google\Chrome\User Data\Default\Extensions , i fixed the issue with the zip file.

The problem now remains that i moved on from using rutorrent and using qbittorrent webui with my seedbox and i cannot for the life of me make it work :)

**Comment by greeneryAO** [comment]
they officially killed it today, I had a backup but it doesn't work loaded, any chance of migrating to v3 many of us were used to it, and others don't work for me in chrome.

**Comment by sejbr** [comment]
I can confirm. I can't find a way to forcefully enable it as of today.

**Comment by Parrakka** [comment]
I managed to get it working by settings the Chrome flags in this thread:
https://www.reddit.com/r/chrome/comments/1lx26ur/for_those_who_want_to_enable_legacy_extensions/

**Comment by greeneryAO** [comment]
thanks, works for now but they'll take it away soon :(

maybe can work until @bogenpirat fixes hopefully

there's also this one that some random made v3 - from reddit

https://www.reddit.com/r/trackers/comments/1hjxvaz/remote_torrent_adder_discontinued_on_chrome/

"I have built one and published to the Google Chrome Store
https://chromewebstore.google.com/detail/holiffefjdehbfhliggafhhlecphpdof?utm_source=item-share-cb

I build from scratch loosely based on the original code idea by
https://github.com/bogenpirat/remote-torrent-adder"

couldn't get it to add one torrent properly though and don't know who he is, rather rta stays alive

**Comment by off3nc3** [comment]
^ Yea that guy is a gem for making a new version of this , shame that @bogenpirat didn't migrate the old one to v3.

**Comment by bogenpirat** [comment]
yeah, as you can tell from my comments in this issue, i've been mostly sticking to aversion tactics with regards to v3. reason being that i've got a day job and a life and a v2->v3 migration asks for a complete rewrite of an, in parts, horrifically shoddy code base. real js/ts devs will attest to that. i'm not much of a javascript/typescript developer, so over the years of working on RTA, i've had more of a focus on quick fixes and dirty feature implementations, more so than maintainability. apart from incompetence, there's sadly also a wealth of different little features, some of which the v3 changes will break. i'm not yet sure how much of it can be salvaged, but i figure that most of the `chrome.webrequest` code in the current background.js file can be scrapped. which sucks, because it is some pretty integral functionality. how much of it can still remain - you'll have to wait and see.

that being said, i've started another rewrite. don't expect anything soon.

**Comment by bogenpirat** [comment]
just to give a brief status update: we're slowly getting places.
- the development in the [mv3 branch](https://github.com/bogenpirat/remote-torrent-adder/tree/feature/manifest-v3) has been... well, progressing. not quickly, but surely.
- currently working: internal torrent downloading, a label/dir selection popup, rutorrent & biglybt support, settings migration from RTAv1
- blocker features that are still missing: the options page to configure RTA (this'll be a LOT of work), more ui support
- there are [automatic builds](https://github.com/bogenpirat/remote-torrent-adder/actions/workflows/build-extension.yml) by github actions. you can download the `chrome-extension-dist` artifact and load it from disk in chrome. but for configuration, you'll need to import an existing RTAv1 dump into localStorage, e.g. with [these](https://github.com/bogenpirat/remote-torrent-adder/wiki/Backup-&-Restore-settings) instructions. but this is completely unstable, may as well skip it. just a first look.

**Comment by greeneryAO** [comment]
thanks bogenpirat :), been surviving on the since taken down instructions for keeping the old one, working great but an official one would be nicer of course

**Comment by greeneryAO** [comment]
can you re-add qbit 4x please, it's not that old, couldn't test torrents because of that, otherwise looks good appreciated.
